ci-setups/php-gitlab.md

```diff
@@ -0,0 +1,140 @@
+# Setup Continuous Integration (CI) on Gitlab for PHP projects
+
+## Introduction
+
+Continuous Integration (CI) in Gitlab is a software development practice that involves automatically integrating code changes from multiple developers into a shared repository on a frequent and ongoing basis. The main goal of CI is to detect and address integration issues early in the development process, ensuring that code changes are tested, validated, and merged into the main codebase as smoothly as possible.
+
+In Gitlab, a CI/CD Pipeline is a set of automated steps that are executed whenever code changes are pushed to the repository. It consists of jobs that can include tasks like building the code, running tests, or performing standards checks for various programming languages, including PHP.
+
+This page will walk you through the steps for setting up a Gitlab Pipeline for performing PHP coding standards checks with PHPCS and Phan for linting and analysis of code.
+
+## Setting up Gitlab Pipeline for project
+
+### Pre-requisites
+
+- An existing or new repository with the PHP project
+- All required configurations done as per [Enforcing tools and config](https://github.com/OsmosysSoftware/dev-standards/blob/main/coding-standards/php.md#enforcing-tools-and-config)
+
+### Creating a Pipeline
+
+GitLab uses a `.gitlab-ci.yml` configuration file in the repository to define the structure and behavior of the CI/CD pipeline. This file specifies the jobs, their dependencies, rules, and scripts to run.
+
+There are two methods that can be followed for creating and configuring a pipeline:
+
+#### Method 1: Using Gitlab UI
+
+1. Open your project repository and click on **Build->Pipeline editor**. Then click on on the **Configure pipeline** button.
+
+    ![Gitlab Pipeline editor](assets/php-gitlab-pipeline-editor.png)
+
+2. A new page will open for editing and creating the `.gitlab-ci.yml` file. Remove all the content that exists in the file, and paste the following code in its place:
+
+    ```yaml
+    stages:
+      - lint
+
+    lint:
+      stage: lint
+      tags:
+        - PHP-8.1
+      rules:
+        # Run pipeline if a commit is made to these branches
+        - if: $CI_COMMIT_BRANCH == "main"
+        # Run pipeline if a merge request is created for these branches
+        - if: $CI_MERGE_REQUEST_TARGET_BRANCH_NAME == "main"
```

Comment:
give sample with multiple target branches.

Reply:
This has been addressed in the latest commit.